core_domain.process_model_execution/src/flow_node_handler/activity_handler/call_activity_handler.ts

```diff
@@ -78,73 +85,86 @@ export class CallActivityHandler extends ActivityHandler<Model.Activities.CallAc
     identity: IIdentity,
   ): Promise<Array<Model.Base.FlowNode>> {
 
-    try {
-      // First we need to find out if the Subprocess was already started.
-      const processInstances = await this.processInstanceRepository.query(identity, {parentProcessInstanceId: flowNodeInstance.processInstanceId});
-
-      const noSubprocessesFound = !(processInstances?.length > 0);
-
-      const matchingSubprocess = noSubprocessesFound
-        ? undefined
-        : processInstances.find((entry): boolean => entry.processDefinitionId === this.callActivity.calledReference);
-
-      let callActivityResult: BpmnEventMessage;
-
-      this.publishActivityReachedNotification(onSuspendToken);
-
-      if (matchingSubprocess == undefined) {
-        // Subprocess not yet started. We need to run the handler again.
-        callActivityResult = await this.executeSubprocess(identity, processTokenFacade, onSuspendToken);
-      } else {
-        // Subprocess was already started. Resume it and wait for the result:
-        callActivityResult = <SystemEventMessage> await this
-          .resumeProcessService
-          .resumeProcessInstanceById(identity, matchingSubprocess.processDefinitionId, matchingSubprocess.processInstanceId);
+    const handlerPromise = new Promise<Array<Model.Base.FlowNode>>(async (resolve: Function, reject: Function): Promise<void> => {
+      try {
+        // First we need to find out if the Subprocess was already started.
+        const processInstances = await this.processInstanceRepository.query(identity, {parentProcessInstanceId: flowNodeInstance.processInstanceId});
+
+        const noSubprocessesFound = !(processInstances?.length > 0);
+
+        const matchingSubprocess = noSubprocessesFound
+          ? undefined
+          : processInstances.find((entry): boolean => entry.processId === this.callActivity.calledReference);
+
+        this.onInterruptedCallback = async (): Promise<void> => {
+          await this.processInstanceManagementService.terminateProcessInstance(identity, this.subProcessInstanceId);
```

Comment:
Hier sollte ein Null-Check hin.

```suggestion
          if (this.subProcessInstanceId) {
            await this.processInstanceManagementService.terminateProcessInstance(identity, this.subProcessInstanceId);
          }
```